### DISCUSSION TITLE
Disabled, Loading and other custom states for Form and Field

### DISCUSSION THREAD

**Discussion by BlackRam-oss** [TanStack/form#1011] [Ideas]
I am currently using React Hook Form, but I wanted to move to TanStack-form because in addition to react, it can be used in various JS Framworks.

I noticed that even in TanStack-form the following options are missing:
* set the entire form to disabled or loading state
* set only a specific field to disabled or loading state, rerendering that field

This option would be appreciated by many developers, who use "Contexts" to change the state of one or more fields from a parent "container".

It would also be nice to set, in addition to the disabled and loading states, other custom states based on your needs. I think that this option is feasible after all, it should be enough to create a T1 object for each form and a T2 object for each field, adding functions to modify these objects and manage the rerenders, and giving the possibility to modify T1 and T2 with ".d.ts" files

**Comment by bruceharrison1984** [comment]
I agree. This ability would negate having to add extra `disabled` props to every reusable form element.
